### ISSUE TITLE
Protected methods gives warning

### ISSUE THREAD

**Issue by olivernybroe** [pestphp/pest-intellij#60]
When accessing protected methods, it marks the calls as an error.
![image](https://user-images.githubusercontent.com/5870441/91630759-75fbdb80-e9d4-11ea-9b4d-a29c2afe1d94.png)

**Comment by adelf** [comment]
Does this work? Dark magic... :)

**Comment by olivernybroe** [comment]
Haha :) This was reported on Discord, a guy thought he made a mistake ;)

**Comment by adelf** [comment]
It can't be fixed programmatically. But we can vote here: https://youtrack.jetbrains.com/issue/WI-25253

I can try to fix that by Php language injection into Php language... but it looks too hacky and ugly.

**Comment by olivernybroe** [comment]
@adelf so I am trying to understand this correctly.

The solution you are linking to would be to disable the inspection instead?

**Comment by adelf** [comment]
For pest test files only.

**Comment by olivernybroe** [comment]
I have written to Olga to see if she has any solutions, else we can hope she can escalate that issue 👍

**Comment by hackel** [comment]
I'm seeing a similar issue when accessing member properties.  e.g.
```php
beforeEach(function () {
    $this->something = 'set up';
});

it('can see setup data', function () {
    expect($this->something)->toBe('set up');
});
```
I am seeing "Member has protected access" on `$this->something`, because in a completely separate PHPUnit test class, I added a protected member called "something" and for some reason IntelliJ is conflating the two.  This might go away once I finally convert that test over to Pest, but for now it causes IntelliJ to mark my entire test file as broken.

**Comment by olivernybroe** [comment]
@hackel yeah this is caused by the exact same issue. 

Hopefully we can find a fix soon

**Comment by olivernybroe** [comment]
So Olga has spend some time with the rest of the PhpStorm team and unfortunately they didn't have a solution for it, just like @adelf said 👍 

Olga created an issue for this and hopefully it will be fixed soon https://youtrack.jetbrains.com/issue/WI-55618

**Comment by RVxLab** [comment]
I'm not experiencing this issue anymore in PhpStorm 2021.2 Beta build 212.4746.20.

![image](https://user-images.githubusercontent.com/46111684/126039089-324a881c-40cb-487b-9053-7d9d5a7a9bf8.png)

**Comment by adelf** [comment]
Yes. They fixed it - https://youtrack.jetbrains.com/issue/WI-55618

**Comment by Lukasss93** [comment]
Hi, I'm experiencing the same problem using the built-in mock function of laravel (via orchestra).

My function "mockC***" works fine but phpstorm warns me with "Member has protected visibility".

Is there a way to fix this "bug"?

![phpstorm64_hOpHK8zfen](https://user-images.githubusercontent.com/4071613/129370265-e4358967-a04c-4447-ab08-94db07cbadc6.png)

**Comment by olivernybroe** [comment]
@Lukasss93 Thanks for reporting this, unfortunately this issue is a phpstorm bug and not something the plugin can solve. 
I have added the case with `test` to our ticket and hopefully they will fix it for us.

re-opening this so others can follow.

**Comment by olivernybroe** [comment]
I have opened a new issue about `test` usage. 
 https://youtrack.jetbrains.com/issue/WI-62239

**Comment by Lukasss93** [comment]
any news about this?

**Comment by olivernybroe** [comment]
@Lukasss93 unfortunately not. We are still blocked by PhpStorm...

**Comment by olivernybroe** [comment]
Closing this issue as there is nothing we can do and it is all on the PhpStorm teams shoulders.
